src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameOfT.cs

```diff
@@ -188,6 +209,10 @@ public class Frame<TContext> : Frame
                 // SetBadRequestState logs the error.
                 SetBadRequestState(ex);
             }
+            catch (IOException ex) when (ex.InnerException is UvException)
```

Comment:
Yeah, what I meant was, let the code above throw and catch it here, then check those conditions up there (`_requestLineStatus` and `StatusCodes`) here and make the appropriate trace call.

I might be missing something, but I don't see the need for the `try`/`catch` above and going through the whole thing with `_requestProcessingStopping` being false. You're ending the loop anyways, just let it throw.